modules/features/app-widgets/src/main/java/com/duchastel/simon/simplelauncher/features/appwidgets/ui/widget/AppWidgetState.kt

```diff
@@ -0,0 +1,15 @@
+package com.duchastel.simon.simplelauncher.features.appwidgets.ui.widget
+
+import androidx.compose.runtime.Immutable
+import com.duchastel.simon.simplelauncher.features.appwidgets.data.WidgetData
+import com.slack.circuit.runtime.CircuitUiState
+
+@Immutable
+data class AppWidgetState(
+    val widgetData: WidgetData,
+    val isLoading: Boolean = false,
+    val error: String? = null,
+    val onRetry: () -> Unit = {},
+    val onRemoveWidget: () -> Unit = {},
+    val onWidgetError: (Throwable) -> Unit = {}
```

Comment:
See my other comment - do not use Throwable! Use typed errors instead that are specific, like a sealed type hierarchy.